sql/account_ecosystem_derived/desktop_clients_daily_v1/query.sql

```diff
@@ -0,0 +1,87 @@
+-- Function to coerce a raw value `v` to the nearest quantile to make fingerprinting
```

Comment:
This method could definitely use some DS scrutiny, but it's something we can evolve later and reprocess, so not critical to get perfect right now.

Reply:
So if I'm understanding this right, we would define the quantile array for each measurement "by hand" (though this in principle could be automated), and then raw values will be replaced by the lower of the two quantile boundaries that the raw value falls between.

This maintains a fixed/absolute scale, which is good. I'm trying to think about what would happen if, for example, the observed distribution for a probe compresses on the low end so that most values start falling between lowest pairs, causing most values to be identical. We can then re-define the quantile array for that measurement to have more resolution in the low end, and since these numbers will still be on the same underlying scale this won't cause problems.

Also, I have to think about this more, but its good that we can pass in an arbitrary array because depending on the underlying distribution for the probe it may be that we do not always want evenly spaced quantiles. Consider a distribution where most of the probability mass is squished into the middle, but still has long, thin tails. If we use deciles in this case, for example, most raw values will be drawn from the middle of the distribution and thus will have similar (or the same) binned values. This will reduce the variance of the binned distribution, potentially to a point where we lose a lot of information. I believe there are other schemes in which the bins are spaced such that they are closer together nearer the center of probability mass. I believe there may be even a general-purpose algorithm that optimizes bin spacing like this given any distribution - I will research.